software/bootcamp-code/image_processor.py

```diff
@@ -20,12 +20,15 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return "[Object: x={}; y={}; size={}; distance={}; exists={}]".format(self.x, self.y, self.size, self.distance, self.exists)
 
-
+# static variable for kernel in image dilation
+kernel=np.array([[0, 1, 0], [1, 1, 1], [0, 1, 0]], dtype=np.uint8)
 # results object of image processing. contains coordinates of objects and frame data used for these results
 class ProcessedResults():
 
     def __init__(self, 
-                balls=[], 
+                balls=[],
+                balls_exist=False,
+                biggest_ball=Object(exists = False),
```

Comment:
Can the biggest ball not be gotten from the balls array? This variable could just be in the main loop but I guess this also works